src/swell/utilities/cylc_runtime.py

```diff
@@ -0,0 +1,243 @@
+
+    def match_platform(self, content: Union[str, dict], platform: str):
+        # Resolve platform-specific entries in the task object
+
+        if isinstance(content, Mapping):
+            if platform in content.keys():
+                content = content[platform]
+            elif 'all' in content.keys():
+                content = content['all']
+
+        return content
+
+    # --------------------------------------------------------------------------------------------------
+
+    def create_new_section(self,
+                           name: Optional[str] = None,
+                           content: Union[str, dict] = ''
+                           ) -> CylcSection:
+        return CylcSection(name, content)
+
+    # --------------------------------------------------------------------------------------------------
+
+    def resolve_model(self, slurm_dict: Mapping) -> dict:
+        ''' Resolve "all" and "model" entries in slurm dictionary '''
+        if 'all' in slurm_dict.keys() and isinstance(slurm_dict['all'], Mapping):
+            slurm_dict = update_dict(slurm_dict, slurm_dict['all'])
+            del slurm_dict['all']
+        if self.model in slurm_dict.keys() and isinstance(slurm_dict[self.model], Mapping):
+            slurm_dict = update_dict(slurm_dict, slurm_dict[self.model])
+
+        for model in get_model_components():
+            if model in slurm_dict.keys():
+                del slurm_dict[model]
+
+        return slurm_dict
+
+    # --------------------------------------------------------------------------------------------------
+
+    def generate_task_slurm_dict(self, slurm_external: Mapping, platform: str) -> Mapping:
+        # Take the external slurm dictionary and merge it with the task's parameters
+        # to get the dict that will be output in the flow.cylc
+
+        slurm_dict = {}
+        if self.slurm is not None:
+            for key, value in self.slurm.items():
+                slurm_dict[key] = self.match_platform(value, platform)
+
+        slurm_globals = slurm_external['slurm_directives_global']
+        slurm_task = {}
+
+        if 'slurm_directives_tasks' in slurm_external.keys():
+            task_directives = slurm_external['slurm_directives_tasks']
+
+            if self.base_name in task_directives:
+                slurm_task = task_directives[self.base_name]
+            if self.scheduling_name in task_directives:
+                slurm_task = task_directives[self.scheduling_name]
+
+        slurm_dict = {'job-name': self.scheduling_name,
+                      **self.resolve_model(slurm_globals),
+                      **self.resolve_model(slurm_dict),
+                      **self.resolve_model(slurm_task)}
+
+        return slurm_dict
+
+    # --------------------------------------------------------------------------------------------------
+
+    def runtime_string(self, experiment_dict: Mapping, slurm_external: Mapping):
```

Comment:
Add typehint for expected output.

```suggestion
    def runtime_string(self, experiment_dict: Mapping, slurm_external: Mapping) -> str:
```